src/TwigComponent/src/Twig/TemplateNameParser.php

```diff
@@ -11,21 +11,24 @@
 
 namespace Symfony\UX\TwigComponent\Twig;
 
+/**
+ * @internal
+ */
 final class TemplateNameParser
```

Comment:
Is this class still necessary?

It's not parsing a template name anymore, but just validating the template name, which already happens in `Twig\Loader\FilesystemLoader::parseName` (in all scenarios in this class, it's just returning the name as-is, so it's just duplicating the validation logic from `FilesystemLoader`)

Reply:
But we cannot remove it because BC i guess.. @kbond ?